### ISSUE TITLE
Automatic seperator for `Flex`

### ISSUE THREAD

**Issue by JAicewizard** [linebender/druid#1344]
Currently there are ways to add separators to `Flex::row/column` but you have to put these in manually and they are just normal widgets with events etc.
On zulip the following ideas ware brought up
 * painter with some kind of layout, this can access state etc. layout options include "I prefer size x" or "I prefer anything bigger than size x", the later doesn't fit well within druid.
 * painter, the caller has to specify the size of the separator and the painter just has to follow

basically rules out ideas:
 * a generic type implementing widget, this was scrapped because it creates more problems than it solves (new type parameter vs less boxing)
